### PR TITLE
Add new commands, fix bugs, and improve .env parsing

### PR DESCRIPTION
**Pull request by bacarndiaye**
New features:
  - Add 'list' command to display configured environments (--json option)
  - Add 'doctor' command for configuration diagnostics
  - Add 'completion' command for shell autocompletion (bash/zsh/fish/powershell)

  Bug fixes:
  - Fix spawn.go: secrets now properly override existing environment variables
  - Fix init_interactive.go: only prompt for .env import when file exists

  Improvements:
  - Improve .env parser to handle escaped quotes within double-quoted strings
  - Add support for escape sequences (\n, \t, \\) in quoted values
  - Add findClosingQuote() helper for escape-aware quote detection

  Tests:
  - Add commands_test.go with tests for new commands
  - Extend envfile_test.go with advanced parsing edge cases

### CODE REVIEW

File: envfile.go [hunk 1]
```diff
@@ -18,22 +25,100 @@ func parseDotEnv(path string) (map[string]string, error) {
...
+	if strings.HasPrefix(val, `"`) {
+		// Find the closing quote, accounting for escaped quotes
+		endQuote := findClosingQuote(val[1:], '"')
+		if endQuote != -1 {
+			// Extract value between quotes, handling escape sequences
+			val = val[1 : endQuote+1]
+			val = strings.ReplaceAll(val, `\"`, `"`)
+			val = strings.ReplaceAll(val, `\\`, `\`)
+			val = strings.ReplaceAll(val, `\n`, "\n")
+			val = strings.ReplaceAll(val, `\t`, "\t")
```
[the remark this answers is not in the document]
Comment (BinSquare) [reply envfile.go:81]: This ai comment is correct - the order was on purpose here.

**Comment by BinSquare** [comment]
I'm not actually sure where the --list command code is added, but I already have below: 
```envmap get --env <name> --all [--raw] [--global] – list all secrets (masked by default); --global iterates all envs.```

Is there a particular usecase you're running into? Otherwise, I'd prefer to keep code surface minimal.

**Comment by BinSquare** [comment]
Overall while I think the intent of this PR is good - it combines multiple discrete changes that we should isolate into separate PR's for a better review.
